src/promptflow-azure/promptflow/azure/_storage/cosmosdb/span.py

```diff
@@ -65,13 +65,36 @@ def persist(self, cosmos_client: ContainerProxy, blob_container_client: Containe
         from azure.cosmos.exceptions import CosmosResourceExistsError
 
         try:
-            return cosmos_client.create_item(body=self.to_dict())
+            return cosmos_client.create_item(body=self.to_cosmosdb_item())
         except CosmosResourceExistsError:
             return
 
     def to_dict(self) -> Dict[str, Any]:
         return {k: v for k, v in self.__dict__.items() if v}
 
+    def to_cosmosdb_item(self, attr_value_truncation_length: int = 8 * 1024):
+        """
+        Convert the object to a dictionary for persistence to CosmosDB.
+        Truncate attribute values to avoid exceeding CosmosDB's 2MB size limit.
+        """
+        item = self.to_dict()  # use to_dict method to get a dictionary representation of the object
+
+        # Serialize the dictionary to a JSON string using separators (",", ":").
+        # This is done to align with the serialization method used by the CosmosDB Python SDK.
+        json_dumped = json.dumps(item, separators=(",", ":"))
+
+        item_size = len(json_dumped.encode("utf-8"))
+        max_size_in_bytes = 2 * 1024 * 1024  # 2MB in bytes
+
+        if item_size > max_size_in_bytes:
```

Comment:
Add a comment to say that the cosmos db's limit is 2MB, and we do truncations for long attributes here.

Reply:
Will remove size condition for clear and fixed behavior for customer.